src/nominatim_db/clicmd/add_data.py

```diff
@@ -65,11 +63,6 @@ def add_args(self, parser: argparse.ArgumentParser) -> None:
     def run(self, args: NominatimArgs) -> int:
         from ..tools import add_osm_data
 
-        with connect(args.config.get_libpq_dsn()) as conn:
-            if is_frozen(conn):
-                print('Database is marked frozen. New data can\'t be added.')
-                return 1
```

Comment:
Your approach only works for Tiger data, so you can't completely get rid of this check. The function also imports other data which needs an unfrozen state.